t/15-shared-plugin-gru.t

```diff
@@ -0,0 +1,41 @@
+#!/usr/bin/env perl
+# Copyright SUSE LLC
+# SPDX-License-Identifier: GPL-2.0-or-later
+
+use Test::Most;
+
+use Test::Warnings ':report_warnings';
+use Test::MockModule;
+use Test::MockObject;
+use FindBin;
+use lib "$FindBin::Bin/lib", "$FindBin::Bin/../external/os-autoinst-common/lib";
+use OpenQA::Test::TimeLimit '5';
+use OpenQA::App;
+use OpenQA::Shared::Plugin::Gru;    # SUT
+
+ok my $gru = OpenQA::Shared::Plugin::Gru->new, 'can instantiate gru plugin';
+
+my $connect_info = {dsn => 'my_dsn', user => 'user', password => 'foo'};
+my $storage = Test::MockObject->new->set_always(connect_info => [$connect_info]);
+my $schema = Test::MockObject->new->set_always(storage => $storage)->set_true('search_path_for_tests');
+my %config = (misc_limits => {minion_job_max_age => 0});
+my $minion = Test::MockObject->new->set_true('on');
+my $log = Test::MockObject->new->set_true('level', 'info');
+my $under = Test::MockObject->new->set_true('to');
+my $routes = Test::MockObject->new->set_always(under => $under);
+my $app
+  = Test::MockObject->new->set_always(schema => $schema)->set_always(config => \%config)->set_always(minion => $minion)
+  ->set_always(log => $log)->set_always(routes => $routes)->set_true('plugin', 'helper');
+OpenQA::App->set_singleton($app);
+my $mock = Test::MockModule->new('OpenQA::Shared::Plugin::Gru');
+$mock->noop('_allow_unauthenticated_minion_stats');
+my $pg = Test::MockObject->new->set_true('dsn', 'username', 'password', 'search_path', 'find');
+my $pg_module = Test::MockModule->new('Mojo::Pg')->redefine(new => $pg);
+ok $gru->register($app, undef), 'can register gru plugin';
+$pg->called_ok('username', 'pg connection initialized with username');
+is(($pg->call_args(0))[1], $connect_info->{user}, 'pg connection username is correct');
+is(($pg->call_args(2))[1], $connect_info->{password}, 'pg connection password is correct');
```

Comment:
I'm wondering about this jump from 0 to 2 in the indexes as password is called directly after user but I guess it is ok.

Reply:
This is also where I think the API of `Test::MockModule` is overshooting the goal. Well have to deal with these indices here when we actually just wanted to check whether properties of a module have eventually been set to some values.

Reply:
Do you have a specific code suggestion you can provide to improve? I wouldn't know otherwise.

Reply:
You can keep it. I just wanted to mention this downside of `Test::MockModule` because you said `Test::MockModule` is the "proper" way compared to just using a sub package.

Reply:
We can also create a local variable and write to that in a mocked module with MockObject/MockModule